sklearn/utils/validation.py

```diff
@@ -374,6 +378,9 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         if force_all_finite:
             _assert_all_finite(array)
 
+    if array.ndim == 1:
```

Comment:
They are probably many errors caused by the same underlying issues. But yes, indeed.

You should be able to just set the ``SingleDimensionWarning`` to "error" though (maybe at the top of the file if it is just for testing).